ConversationTriggers

aaronschachter opened the pull request:
Queries the `defaultTopicTrigger` Contentful type.

DFurnes commented on src/schema/gambitContent.js:
```diff
@@ -139,9 +139,25 @@ const typeDefs = gql`
     ${broadcastFields}
   }
 
+  # A conversation trigger, used to change topic or answer a FAQ.
```
Before merging this, let's just update these to the new syntax, e.g:

```diff
- # A conversation trigger, used to change topic or answer a FAQ.
+ "A conversation trigger, used to change topic or answer a FAQ."
  type ConversationTrigger {
```

aaronschachter commented:
Rebased with `master` and updated new schema syntax. This is the query Gambit Conversations will execute to fetch the Rivescript triggers we define in Contentful (deprecating querying the Gambit Content ` GET /defaultTopicTriggers` endpoint)

 ```
{
  conversationTriggers {
    id
    trigger
    reply
    topic {
      id
      name
      ... on AutoReplySignupTopic {
        campaign {
          id
          endDate
        }
      }
      ... on PhotoPostTopic {
        campaign {
          id
          endDate
        }
      }
      ... on TextPostTopic {
        campaign {
          id
          endDate
        }
      }
    }
  }
}

```

Response:
```
{
  "data": {
    "conversationTriggers": [
      {
        "id": "2KJhNbHWy42mwwSmQaIC2g",
        "trigger": "supplies",
        "reply": "There are almost 600,000 people experiencing homelessness in America. About 1/4 of them are teens or children.\n\nThe solution is simple: join 6 million DoSomething members by providing daily essentials (like toothbrushes, toothpaste, soap, and nonperishable foods) to a local homeless shelter.\n\nOnce you’ve created your bags filled with daily essentials to donate to a shelter, take a pic with them to prove it! Then text START to share it with us (and enter to win a $5k scholarship)!",
        "topic": {
          "id": "1SJeHKZTCs4QIIAy4MoQMq",
          "name": "Supplies, Sealed, Delivered - Donation Bags",
          "campaign": {
            "id": 8308,
            "endDate": "2019-03-31T00:00:00Z"
          }
        }
      },
      {
        "id": "5mMWVfjbkAYwmuKYeOOIey",
        "trigger": "card",
        "reply": "Thanks for joining Not-So-Secret Admirer! Valentine’s Day is the perfect time to give your friends the confidence to achieve their dreams.\n\nUsing our template, give your friend a V-Day card! Include “text LOVE to 38383” on the inside of the card so they can get tips to take their passions to the next level and the chance to apply for Taco Bell Foundation’s Live Más Scholarship.\n\nhttps://downloads.ctfassets.net/81iqaqpfd8fy/3JwHTfrNfiCqwuMsskswM0/bffbcceef2f5c1fdcf1a424c8594620b/Not-So-Secret_Admirer_Final_Cards.pdf\n\nOnce you’ve created your card, text START to send us a pic of you, your friend, and the card! (You'll be entered for a chance to win a $5,000 scholarship from DoSomething!)",
        "topic": {
          "id": "Wo3a7CdlIGKy2cauieWky",
          "name": "Not-So-Secret Admirer - V-Day Card",
          "campaign": {
            "id": 8292,
            "endDate": "2019-02-21T00:00:00Z"
          }
        }
      },
      {
   ...
```
